### PR TITLE
[Monitor Query] Typing updates

### PR DESCRIPTION
**Pull request by pvaneck**
Regenerated with latest autorest and updated some samples. 
Changed the model types for `status` to use Literal with the corresponding enum values to improve typing.

Closes: #32377, #32379

### CODE REVIEW

File: sdk/monitor/azure-monitor-query/README.md [hunk 1]
```diff
@@ -147,12 +147,14 @@ query = """AzureActivity | take 5"""
 
 try:
     response = client.query_resource(os.environ['LOGS_RESOURCE_ID'], query, timespan=timedelta(days=1))
-    if response.status == LogsQueryStatus.PARTIAL:
+    if isinstance(response, LogsQueryResult):
```
Comment (kristapratico) [inline sdk/monitor/azure-monitor-query/README.md:150]: Is it possible to use a [tagged union](https://mypy.readthedocs.io/en/stable/literal_types.html#tagged-unions) here so we don't have to do isinstance checks for typing?

Comment (pvaneck) [reply sdk/monitor/azure-monitor-query/README.md]: Thanks for the pointers. Updated the `status` values to user Literals with the corresponding enum values. Now, the `response.status` checks can narrow down the typing and we no longer need the `isinstance` checks.

File: sdk/monitor/azure-monitor-query/azure/monitor/query/_exceptions.py [hunk 2]
```diff
@@ -27,15 +27,14 @@ class LogsQueryError:
...
-    status: LogsQueryStatus
+    status: Literal[LogsQueryStatus.FAILURE]
     """Status for error item when iterating over list of results. Always "Failure" for an instance of a
     LogsQueryError."""
 
     def __init__(self, **kwargs: Any) -> None:
         self.code = kwargs.get("code", "")
         self.message = kwargs.get("message", "")
         self.details = kwargs.get("details", None)
-        self.status = LogsQueryStatus.FAILURE
```
Comment (kristapratico) [inline sdk/monitor/azure-monitor-query/azure/monitor/query/_exceptions.py:38]: I think we still need the attr defined, can you double-check?

File: sdk/monitor/azure-monitor-query/samples/async_samples/sample_batch_query_async.py [hunk 3]
```diff
@@ -54,20 +54,20 @@ async def logs_batch_query():
             results = await client.query_batch(requests)
 
             for res in results:
-                if res.status == LogsQueryStatus.FAILURE:
-                    # This will be a LogsQueryError
-                    print(res)
+                if res.status == LogsQueryStatus.SUCCESS:
```
Comment (kristapratico) [inline sdk/monitor/azure-monitor-query/samples/async_samples/sample_batch_query_async.py:57]: I like that you moved Success above Failure :)

File: sdk/monitor/azure-monitor-query/azure/monitor/query/_exceptions.py (hunk 2 above)
Comment (pvaneck) [reply sdk/monitor/azure-monitor-query/azure/monitor/query/_exceptions.py]: Ah good catch! Re-added those back.

**Review by kristapratico — APPROVED** [review]

**Pull request merged**